### PR TITLE
Add MatPES GGA and r2SCAN static makers

### PR DESCRIPTION
**Pull request by janosh**
Related PR that defined the input set: https://github.com/materialsproject/pymatgen/pull/3254.

### CODE REVIEW

**Comment by mkhorton** [comment]
Your comment over in pymatgen encouraged me to check this PR out :)

What's the story behind the change to `PBE_54` in `MPGGA`? Is this going to be incompatible with MP data, or is MP data being refreshed too?

**Comment by mkhorton** [comment]
Thanks, I'll leave a comment in #504. I understand this PR is for something else, but I noticed the `MPGGA` set was being changed too.

**Comment by janosh** [comment]
Confused by the test failure here. It complains about `LREAL` being `'Auto'` while `False` is expected. It should be `False` since https://github.com/materialsproject/pymatgen/commit/8f9bbd6884052e3984c3fe6fd16c56e6b2ce7f99 which was released in pymatgen [v2023.9.25](https://github.com/materialsproject/pymatgen/releases/tag/v2023.9.25) which is also the version installed in atomate2 CI...

File: src/atomate2/vasp/sets/matpes.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+@dataclass
+class MatPesGGAStaticSetGenerator(StaticSetGenerator):
+    """Class to generate MP-compatible VASP GGA static input sets."""
+
+    config_dict: dict = field(default_factory=lambda: _BASE_MATPES_PBE_STATIC_SET)
+    auto_ismear: bool = False
+    auto_kspacing: bool = True
+    user_incar_settings: dict = field(
+        # ensure _set_kspacing doesn't override input set ISMEAR
+        default_factory=lambda: {"ISMEAR": 0, "SIGMA": 0.05}
```
Comment (utf) [inline src/atomate2/vasp/sets/matpes.py:42]: Is this needed now?

File: src/atomate2/vasp/sets/matpes.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+@dataclass
+class MatPesMetaGGAStaticSetGenerator(StaticSetGenerator):
+    """Class to generate MP-compatible VASP GGA static input sets."""
+
+    config_dict: dict = field(default_factory=lambda: _BASE_MATPES_PBE_STATIC_SET)
+    auto_ismear: bool = False
+    auto_kspacing: bool = True
+    user_incar_settings: dict = field(
+        # ensure _set_kspacing doesn't override input set ISMEAR
+        default_factory=lambda: {"ISMEAR": 0, "SIGMA": 0.05}
```
Comment (utf) [inline src/atomate2/vasp/sets/matpes.py:55]: Is this needed?

**Review by utf — COMMENTED** [review]
Two things I noticed.

File: src/atomate2/vasp/sets/matpes.py (hunk 2 above)
Comment (janosh) [reply src/atomate2/vasp/sets/matpes.py:55]: No, I'll remove.

**Pull request merged**